docs/guide/migration.md

```diff
@@ -1,147 +1,77 @@
-
-If using ESM for SSR isn't possible in your project, you can set `legacy.buildSsrCjsExternalHeuristics` to generate a CJS bundle using the same externalization strategy of Vite v2.
-
-Also [`build.rollupOptions.output.inlineDynamicImports`](https://rollupjs.org/guide/en/#outputinlinedynamicimports) now defaults to `false` when `ssr.target` is `'node'`. `inlineDynamicImports` changes execution order and bundling to a single file is not needed for node builds.
+...
 
 ## General Changes
 
-- JS file extensions in SSR and lib mode now use a valid extension (`js`, `mjs`, or `cjs`) for output JS entries and chunks based on their format and the package type.
-- Terser is now an optional dependency. If you are using `build.minify: 'terser'`, you need to install it.
-  ```shell
-  npm add -D terser
-  ```
-
-### `import.meta.glob`
-
-- [Raw `import.meta.glob`](features.md#glob-import-as) switched from `{ assert: { type: 'raw' }}` to `{ as: 'raw' }`
-- Keys of `import.meta.glob` are now relative to the current module.
-
-  ```diff
-  // file: /foo/index.js
-  const modules = import.meta.glob('../foo/*.js')
-
-  // transformed:
-  const modules = {
-  -  '../foo/bar.js': () => {}
-  +  './bar.js': () => {}
-  }
-  ```
+### Importing CSS as a string
 
-- When using an alias with `import.meta.glob`, the keys are always absolute.
-- `import.meta.globEager` is now deprecated. Use `import.meta.glob('*', { eager: true })` instead.
+In Vite 3, importing the default export of a `.css` file could introduce a double loading of CSS.
 
-### WebAssembly Support
-
-`import init from 'example.wasm'` syntax is dropped to prevent future collision with ["ESM integration for Wasm"](https://github.com/WebAssembly/esm-integration).
-You can use `?init` which is similar to the previous behavior.
-
-```diff
--import init from 'example.wasm'
-+import init from 'example.wasm?init'
-
--init().then((exports) => {
-+init().then(({ exports }) => {
-  exports.test()
-})
+```ts
+import cssString from './global.css'
 ```
 
-### Automatic https Certificate Generation
-
-A valid certificate is needed when using `https`. In Vite v2, if no certificate was configured, a self-signed certificate was automatically created and cached.
-Since Vite v3, we recommend manually creating your certificates. If you still want to use the automatic generation from v2, this feature can be enabled back by adding [@vitejs/plugin-basic-ssl](https://github.com/vitejs/vite-plugin-basic-ssl) to the project plugins.
-
-```js
-import basicSsl from '@vitejs/plugin-basic-ssl'
+Since `.css` file will be emitted, but it's likely that the CSS string will also be used by the application code, for example, injected by the framework runtime. From Vite 4, the `.css` default export [has been deprecated](https://github.com/vitejs/vite/issues/11094). The `?inline` query suffix modifier needs to be used in this case, as that doesn't emit the imported `.css` styles.
```

Comment:
I just sent a PR to suggest different wording for this sentence in the changelog: https://github.com/vitejs/vite/pull/11224